pep.rst

```diff
@@ -194,7 +227,7 @@ For example:
 Valid Tag Names
 ---------------
 
-The tag name can be any **undotted** name that isn't already an existing valid
+The tag name can be any *undotted* name that isn't already an existing valid
```

Comment:
Out of curiosity, I'd love to understand the instinct here, particularly in light of PEP614's relaxing of decorator syntax. I could imagine wanting, at minimum, dots:

```python
import somelibrary

x = somelibrary.sometag"..."
```

Beyond dots, JavaScript allows tags to be arbitrary expressions. Perhaps that'd get too awkward in python?:

```python
x = (lambda *args: args)"..."
```

Excited about this PEP!

Reply:
I will let @jimbaker address the motivation here. Glad you're excited about the PEP, lots we can do with this.

Reply:
@davepeck in principle we could allow both dotted names and arbitrary expressions, since both are currently invalid syntax. This was briefly mentioned here in the bikeshed issue, https://github.com/jimbaker/tagstr/issues/3#issue-1222506785

Given how much namespaces are used in current code - and the last admonition of `import this`/Zen of Python, "Namespaces are one honking great idea -- let's do more of those!", perhaps we should relax this constraint; and possibly the use of parenthesized expression as well (to keep it reasonable, at least for human readers).

Reply:
FWIW I would also vote for a more relaxed syntax here. Dotted names at least, or even PEP 614-compatible syntax.